File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = {i:-1 for i in range(256)}
        l = 0
        r = 0
        n = len(s)
        maxi = 0
        while(r<n):
            m = d[ord(s[r])]
            if(m!=-1):
                l = max(m+1,l)
            d[ord(s[r])] = r
            maxi = max(maxi,r-l+1)
            r+=1
        return maxi

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        int left = 0, right = 0;
        int ans = 0;
        unordered_map<char,int> freqCount;
        
        for(right = 0; right < n; right++){
            freqCount[s[right]]++;
            while(freqCount[s[right]] > 1){
                freqCount[s[left]]--;
                left++;
            }
            ans = max(ans, right - left + 1);
        }
        return ans;
    }
};